[Kushal Das] iP

kushaldas27 opened the pull request:
# Elon 

>"While it is true that Kasparov is almost as good at playing chess as my iPhone, he is otherwise an idiot" - Elon Musk ([source](https://x.com/elonmusk/status/1584533444444332033?lang=en))

Elon frees your mind of having to remember things you need to do. It's,
- text-based
- easy to learn
- FAST SUPER FAST to use

All you need to do is,

1. download it 
2. add your tasks.
3. let it manage your tasks for you 😉
4. And it is **FREE!**

Features:
- [X]  Managing tasks
- [ ] Managing deadlines (coming soon)
- [ ] Reminders (coming soon)

If you are a Java programmer, you can use it to practice Java too. Here's the main method:
```java
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

kushaldas27 committed:
Rename chatbot to Elon

kushaldas27 committed:
Add rename, greet and exit

kushaldas27 committed:
Add Echo and Bye functionality

kushaldas27 committed:
Added List functionality

kushaldas27 committed:
Add top horizontal lines

kushaldas27 committed:
Add mark as done and drawLine

kushaldas27 committed:
Add ToDo, Deadline and Event Classes

kushaldas27 committed:
Add testing

kushaldas27 committed:
Refactor code and create Action class

kushaldas27 committed:
Exception Handling

kushaldas27 committed:
Add delete task

kushaldas27 committed:
Add save file

kushaldas27 committed:
Merge pull request #1 from kushaldas27/branch-level-7

Add save file

kushaldas27 committed:
Add LocalDate to respective Tasks

kushaldas27 committed:
Merge pull request #2 from kushaldas27/branch-level-8

Add LocalDate to respective Tasks

kushaldas27 committed:
Refactor into separate classes

kushaldas27 committed:
Merge pull request #3 from kushaldas27/branch-level-8

Refactor into separate classes

kushaldas27 committed:
Create Elon package

kushaldas27 committed:
Merge pull request #4 from kushaldas27/branch-level-8

Create Elon package

kushaldas27 committed:
Merge pull request #5 from kushaldas27/add-gradle-support

Add gradle support

jayjay19630 commented on src/main/java/Elon/Parser.java:
```diff
@@ -0,0 +1,93 @@
+    public String[] nextInput(Scanner scanner) {
+        return scanner.nextLine().split(" ");
+    }
+    public void Parse() throws ElonException{
```
Since this is a method, please use camelCase for method names, i.e. `parse` instead of `Parse`.

jayjay19630 commented on src/main/java/Elon/Deadline.java:
```diff
@@ -0,0 +1,25 @@
+    protected LocalDate by;
+    public Deadline(String description, boolean isDone, LocalDate by) {
+        super(description, isDone);
+        this.by =by;
```
Please use whitespace between operands, i.e. `this.by = by` instead of `this.by =by`.

jayjay19630 commented on src/main/java/Elon/Storage.java:
```diff
@@ -0,0 +1,86 @@
+        boolean isDone = elements[1].equals("1");
+        String description = elements[2];
+
+        switch (type) {
+            case "T":
+                return new ToDo(description, isDone);
+            case "D":
+                LocalDate by = LocalDate.parse(elements[3]);
+                return new Deadline(description, isDone, by);
+            case "E":
+                LocalDate start = LocalDate.parse(elements[3]);
+                LocalDate end = LocalDate.parse(elements[4]);
+                return new Event(description, isDone, start, end);
+            default:
+                throw new ElonException("Invalid task format.");
+        }
```
Since this is a switch case statement, please avoid indentation for `case`, as per the coding standard:

<img width="987" alt="Screenshot 2024-09-03 at 11 45 02 AM" src="https://github.com/user-attachments/assets/92cfeb17-b1ac-4af0-9055-5f89621f27f9">

jayjay19630 commented on src/main/java/Elon/Ui.java:
```diff
@@ -0,0 +1,164 @@
+        drawLine();
+    }
+
+    public void numOfTasks(int size) {
```
Since this is a method, should we make this method name a verb?

kushaldas27 committed:
Add junit tests

kushaldas27 committed:
Merge pull request #6 from kushaldas27/heads/A-Gradle

Add junit tests

kushaldas27 committed:
Add javadoc comments

kushaldas27 committed:
Add find task

kushaldas27 committed:
Merge pull request #7 from kushaldas27/branch-level-9

Branch level 9

Wnayar commented on src/main/java/elon/Deadline.java:
```diff
@@ -0,0 +1,53 @@
+                + by.format(DateTimeFormatter.ofPattern("MMM d yyyy")) + ")";
+    }
+
+    /**
+     * Returns a string suitable for saving to a file, representing the Deadline task.
+     * The format is "D | status | description | deadlineDate",
+     * where status is "1" if the task is done, or "0" if not done.
+     * The deadline date is represented in default LocalDate format.
+     *
+     * @return the string representation of the Deadline task for saving to a file
+     */
```
Great descriptive headers for comments!

Wnayar commented on src/main/java/elon/Deadline.java:
```diff
@@ -0,0 +1,53 @@
+     * @return the string representation of the Deadline task for saving to a file
+     */
+    @Override
+    public String toFileString() {
```
I think this name is not so easy to understand at first glance. I only understood it after reading the header comment, Perhaps it could be changed to parseIntoFileLine. Or maybe something better, my suggested name may not be the best.

Wnayar commented on src/main/java/elon/Parser.java:
```diff
@@ -0,0 +1,120 @@
+                int index = Integer.parseInt(inputArr[1]) - 1;
+                ui.deleteTask(index, list);
+                try {
+                    storage.saveFile(list);
+                } catch (IOException e) {
+                    System.out.println(e);
+                }
+                inputArr = this.nextInput(scanner);
+                continue;
+            } else if (inputArr[0].equals("find")) {
+                String keyword = String.join(" ", Arrays.copyOfRange(inputArr, 1, inputArr.length));
+                ArrayList<Task> matchingTasks = list.findTasks(keyword);
+                ui.showMatchingTasks(matchingTasks);
+                inputArr = this.nextInput(scanner);
+                continue;
+            } else {
+                if (inputArr[0].equals("todo")) {
+                    try {
+                        ui.addToDo(inputArr, list);
+                        storage.saveFile(list);
+                    } catch (ElonException | IOException e) {
+                        System.out.println(e);
+                    }
+                } else if (inputArr[0].equals("deadline")) {
+                    try {
+                        ui.addDeadline(inputArr, list);
+                        storage.saveFile(list);
+                    } catch (ElonException | IOException e) {
+                        System.out.println(e);
+                    }
+                } else if (inputArr[0].equals("event")) {
+                    try {
+                        ui.addEvent(inputArr, list);
+                        storage.saveFile(list);
+                    } catch (ElonException | IOException e) {
+                        System.out.println(e);
+                    }
+                } else {
+                    throw new ElonException("Error. Invalid command.");
+                }
```
Have you considered using a switch statement? I don't think a switch statement is compulsory, but it makes readability a lot easier. But nice work for accounting for all the cases!

Wnayar commented on src/main/java/elon/Parser.java:
```diff
@@ -0,0 +1,120 @@
+    }
+    public void Parse() throws ElonException{
+        while (!inputArr[0].equals("bye")) {
+            if (inputArr[0].equals("list")) {
```
Have you considered using an enumeration to match the inputs. Enums also play nice into using switch statements in my comment below. By using enumerations, it will improve readability as at one glance you be able to see all commands currently and allows for easier maintenance. But nice work overall!

Wnayar commented on src/main/java/elon/Ui.java:
```diff
@@ -0,0 +1,259 @@
+     * Draws a horizontal line in the console for separation.
+     */
+    public void drawLine() {
+        System.out.println("\t -------------------------------------------------------");
```
Nice work abstracting out the line break into a variable.

Wnayar reviewed (commented):
LGTM. Just some nits related to CS2103T code quality and coding standards. Awesome Job!

kushaldas27 committed:
Initial GUI

tzuennn commented on src/main/java/elon/Deadline.java:
```diff
@@ -0,0 +1,53 @@
+ */
+public class Deadline extends Task {
+    protected boolean isDone;
+    protected LocalDate by;
```
I think the naming can be more descriptive! If the program is expanded in the future to include multiple users. "by" could also refer to the name of the person who completed the task.

tzuennn commented on src/main/java/elon/Deadline.java:
```diff
@@ -0,0 +1,53 @@
+     */
+    public Deadline(String description, boolean isDone, LocalDate by) {
+        super(description, isDone);
+        this.by =by;
```
Try to include a space after the "=" to make it more readable!

tzuennn commented on src/main/java/elon/Elon.java:
```diff
@@ -0,0 +1,55 @@
+     *
+     * @throws ElonException if an error occurs during command processing
+     */
+    private void Run() throws ElonException {
```
Try your best to follow the naming conventions! Methods should start with a lowercase letter and use camelCase.

tzuennn commented on src/main/java/elon/Storage.java:
```diff
@@ -0,0 +1,115 @@
+     * @throws ElonException if the line format is invalid or cannot be parsed
+     */
+    private Task parseTask(String line) throws ElonException {
+        String[] elements = line.split(" \\| ");
```
There is a clear distinction between single and multivariable naming, nice!

tzuennn commented on src/main/java/elon/Storage.java:
```diff
@@ -0,0 +1,115 @@
+     */
+    public void saveFile(TaskList list) throws IOException {
+        try(FileWriter fileWriter = new FileWriter(this.file)) {
+            int i = 0;
```
The scope of i is not the best. Since its only used within the while loop, its good practice to limit the scope of the loop variables to the smallest possible block to improve readability, and more importantly to prevent accidental misuse of outside of the intended loop.

tzuennn reviewed (commented):
Overall good job! Slight improvement is to separate the classes into more meaningful packages.

kushaldas27 committed:
Fix coding standard violations

kushaldas27 committed:
Refactor into commands and improve GUI

kushaldas27 committed:
Merge pull request #9 from kushaldas27/branch-level-10

Branch level 10

kushaldas27 committed:
Fix exception handling

kushaldas27 committed:
Merge pull request #10 from kushaldas27/branch-level-10

Fix exception handling

kushaldas27 committed:
Add assertions

kushaldas27 committed:
Merge pull request #11 from kushaldas27/heads/A-Assertions

Add assertions

kushaldas27 committed:
Add JavaDoc for commands

kushaldas27 committed:
Revert "Add JavaDoc for commands"

This reverts commit e88d9856c7aa3e0da1a53f852f60ee571e634929.

kushaldas27 committed:
Reapply "Add JavaDoc for commands"

This reverts commit 161a5522d2f9445f5b990b75572ed200b0de700f.

kushaldas27 committed:
Merge pull request #12 from kushaldas27/branch-A-JavaDoc

Branch a java doc

kushaldas27 committed:
Elon application: Add Javadoc comments for main classes

The Elon application serves as a task management chatbot, providing
functions for user interaction, command parsing, and task handling.
Javadoc comments have been added to the `Elon`, `ElonException`,
`Parser`, and `Ui` classes to improve code documentation and
understanding.

The added Javadocs include:

* `Elon`: Describes the main entry point, task list management,
  and user input handling.
* `ElonException`: Documents the custom exception used for handling
  errors related to the Elon chatbot.
* `Parser`: Provides details on how user commands are parsed
  and processed, including error handling for invalid inputs.
* `Ui`: Outlines the user interface functions that facilitate
  interaction with the task list and display messages to the user.

These changes enhance code readability and maintainability, making
it easier for future developers to understand the purpose and
functionality of each class.

kushaldas27 committed:
Merge pull request #13 from kushaldas27/branch-A-Assertions

Elon application: Add Javadoc comments for main classes

kushaldas27 committed:
Remove tests and add checkstyle

kushaldas27 committed:
Merge pull request #14 from kushaldas27/branch-A-CodingStandard

Remove tests and add checkstyle

kushaldas27 committed:
no message

kushaldas27 committed:
Revert "no message"

This reverts commit eb619e50b5abc7b55631a1352e4f509a2c6258d1.

kushaldas27 committed:
Fix checkstyle errors

kushaldas27 committed:
Merge pull request #15 from kushaldas27/branch-A-CodingStandard

Branch a coding standard

kushaldas27 committed:
Add junit tests

kushaldas27 committed:
Merge pull request #16 from kushaldas27/branch-A-CodingStandard

Add junit tests

kushaldas27 committed:
Merge pull request #17 from kushaldas27/branch-A-CodeQuality

Branch a code quality

kushaldas27 committed:
Update user guide

kushaldas27 committed:
Update exit command

kushaldas27 committed:
Merge pull request #18 from kushaldas27/branch-A-CodingStandard

Branch a coding standard

kushaldas27 committed:
Add acknowledgements

kushaldas27 committed:
Merge pull request #19 from kushaldas27/branch-A-CodingStandard

Add acknowledgements
